pkg/cloudprovider/provider/anexia/provider.go

```diff
@@ -587,3 +589,23 @@ func updateMachineStatus(machine *clusterv1alpha1.Machine, status anxtypes.Provi
 
 	return nil
 }
+
+func anexiaErrorToTerminalError(err error, msg string) error {
+	var httpError api.HTTPError
+	if errors.As(err, &httpError) && (httpError.StatusCode() == http.StatusForbidden || httpError.StatusCode() == http.StatusUnauthorized) {
```

Comment:
Checking for Unauthorized is fine, but I'm not sure about Forbidden as that is not about invalid credentials but e.g. used by some APIs to reply to rate limited users (even though there is an explicit status code for that ..)

Did you see terminal error Engine responses with Forbidden status?

Same question applies below.

Reply:
> Did you see terminal error Engine responses with Forbidden status?

Not in relation to this issue. But wouldn't a wrongly configured access token (e.g. missing vm create permissions) lead to a `Forbidden` response? And wouldn't this also be a kind of "invalid credentials"?